plugins/polio/js/src/domains/VaccineModule/StockManagement/StockVariation/Modals/CreateEditEarmarked.tsx

```diff
@@ -76,7 +80,17 @@ export const CreateEditEarmarked: FunctionComponent<Props> = ({
         titleMessage,
     )} ${formatMessage(MESSAGES.earmarked)}`;
     const allowConfirm = formik.isValid && !isEqual(formik.touched, {});
-
+    const { data: vrfList } = useGetVrfListByRound(selectedRound?.original?.id);
+    const quantityOrdered = vrfList?.reduce(
+        (acc, vrf) => acc + vrf.quantities_ordered_in_doses,
+        0,
+    );
+    // https://bluesquare.atlassian.net/browse/POLIO-1924
+    useEffect(() => {
+        if (quantityOrdered && quantityOrdered > 0 && !earmark?.id) {
```

Comment:
I would put `!earmark?.id` in a const `isNewEarmark` or some name in that style and `quantityOrdered && quantityOrdered > 0 ` in another so this line can be refactored as something like: 

```suggestion
        if (hasQuantityOrdered && isNewEarmark) {
```